pkg/controller/framework/shared_informer.go

```diff
@@ -64,26 +65,31 @@ func NewSharedInformer(lw cache.ListerWatcher, objType runtime.Object, resyncPer
 // TODO: create a cache/factory of these at a higher level for the list all, watch all of a given resource that can
 // be shared amongst all consumers.
 func NewSharedIndexInformer(lw cache.ListerWatcher, objType runtime.Object, resyncPeriod time.Duration, indexers cache.Indexers) SharedIndexInformer {
-	sharedIndexInformer := &sharedIndexInformer{
-		processor:        &sharedProcessor{},
-		indexer:          cache.NewIndexer(DeletionHandlingMetaNamespaceKeyFunc, indexers),
-		listerWatcher:    lw,
-		objectType:       objType,
-		fullResyncPeriod: resyncPeriod,
+	return newCoreInformer(lw, objType, resyncPeriod, indexers)
+}
+
+func newCoreInformer(lw cache.ListerWatcher, objType runtime.Object, resyncPeriod time.Duration, indexers cache.Indexers) *coreInformer {
+	indexer := cache.NewIndexer(DeletionHandlingMetaNamespaceKeyFunc, indexers)
+	s := &coreInformer{
+		indexer: indexer,
+		cfg: &Config{
+			Queue:            cache.NewDeltaFIFO(cache.MetaNamespaceKeyFunc, nil, indexer),
+			ListerWatcher:    lw,
+			ObjectType:       objType,
+			FullResyncPeriod: resyncPeriod,
+			RetryOnError:     false,
+		},
 	}
-	return sharedIndexInformer
+	s.cfg.Process = s.Process
+	s.controller = New(s.cfg)
+	return s
 }
 
-type sharedIndexInformer struct {
+type coreInformer struct {
 	indexer    cache.Indexer
 	controller *Controller
-
-	processor *sharedProcessor
-
-	// This block is tracked to handle late initialization of the controller
-	listerWatcher    cache.ListerWatcher
-	objectType       runtime.Object
-	fullResyncPeriod time.Duration
+	handlers   []ResourceEventHandler
```

Comment:
Single `ResourceEventHandler`.  More that one should only be allowed from a `SharedIndexInformer`.  If anyone makes an `UnsafeUnionEventHandler`, it should be a very explicit choice that clearly evokes the danger of what they're doing.